vector_setters.go

```diff
@@ -32,108 +33,319 @@ func setPrimitive[T any](vec *vector, rowIdx C.idx_t, v T) {
 	xs[rowIdx] = v
 }
 
-func (vec *vector) setTS(t Type, rowIdx C.idx_t, val any) {
-	v := val.(time.Time)
+// TODO: support Decimal here and bool and hugeint
+func setNumeric[S any, T numericType](vec *vector, rowIdx C.idx_t, val S) error {
+	var fv T
+	switch v := any(val).(type) {
+	case uint8:
+		fv = T(v)
+	case int8:
+		fv = T(v)
+	case uint16:
+		fv = T(v)
+	case int16:
+		fv = T(v)
+	case uint32:
+		fv = T(v)
+	case int32:
+		fv = T(v)
+	case uint64:
+		fv = T(v)
+	case int64:
+		fv = T(v)
+	case uint:
+		fv = T(v)
+	case int:
+		fv = T(v)
+	case float32:
+		fv = T(v)
+	case float64:
+		fv = T(v)
+	case bool:
+		if v {
+			fv = 1
+		} else {
+			fv = 0
+		}
+	case *big.Int:
+		if v == nil {
+			return castError(reflect.TypeOf(val).String(), reflect.TypeOf(fv).String())
+		}
+		if v.IsUint64() {
+			fv = T(v.Uint64())
+		} else {
+			fv = T(v.Int64())
+		}
```

Comment:
Let's remove these cases, as we currently do not use them.

Reply:
The bool and bigint case? Sure, I mostly added all the cases because I thought it would be nice to support casting from as many types as possible, but I can reduce the amount of cases if you want. Should I just reduce the supported types to what was possible before?

Reply:
I think either 
1. We add code testing all these new cases or
2. We remove them for now and potentially add them later with respective tests.

My main issue is that our tests do not trigger these paths (afaik).

Reply:
nope, I dont think these paths are tested, but the tests seem trivial. Ill just remove it for now, can always be added later.